src/Microsoft.ApplicationInsights/DataContracts/AvailabilityTelemetry.cs

```diff
@@ -17,7 +17,7 @@ public sealed class AvailabilityTelemetry : ITelemetry, ISupportProperties, ISup
     {
         internal const string TelemetryName = "Availability";
 
-        internal readonly string BaseType = typeof(AvailabilityData).Name;
+        internal readonly string BaseType = nameof(AvailabilityData);
```

Comment:
Consider "const" instead of "readonly". Then there's no runtime cost.

Reply:
very valid feedback, now it can be a compile time const shared across instances.